### DISCUSSION TITLE
openDTU Daten abrufen mit PHP

### DISCUSSION THREAD

**Discussion by adalbertgreben** [tbnobody/OpenDTU#1669] [General]
Ich bin neu hier. Ich hoffe das ist die richtige Diskussionsrunde.
Ich entwickle HTML/JS und PHP.
Ich würde gerne wissen wie man die Daten (zB: Leistung: Wert und Einheit) mit PHP $url = "http://192.168.178.30";   $response = file_get_contents($url); abrufen kann. Im info/console werden alle an HM-800  gesendete und empfangene Daten angezeigt.Ich möchte mir eigene Grafiken dazu erstellen. Danke für jede Antwort.
Albert

**Comment by pieterhollander** [comment]
If you want to get the OpenDTU metrics live to draw your own graphs, you can set up a websocket connection to `http://YOUR_IP_ADDRESS/livedata`.  The values will refresh every few seconds, depending on how you've configured OpenDTU.

For your inspiration, this is how I'm doing something similar in Golang, but instead to log the livedata to a database:

```go
func main() {
	dbConnStr := (os.Getenv("DB_URL"))
	// Connect to PostgreSQL
	db, err := sql.Open("postgres", dbConnStr)
	if err != nil {
		log.Fatal(err)
	}
	defer db.Close()

	// Create tables if they don't exist
	createTables(db)

	// Get WebSocket URL from environment variable
	RemoteURL := os.Getenv("REMOTE_URL")
	wsURL := "ws://" + RemoteURL + "/livedata"
	if wsURL == "" {
		log.Fatal("WEBSOCKET_URL environment variable is not set.")
	}

	// Establish WebSocket connection
	c, _, err := websocket.DefaultDialer.Dial(wsURL, nil)
	if err != nil {
		log.Fatal(err)
	}
	defer c.Close()

	// Subscribe to the WebSocket feed
	if err := c.WriteMessage(websocket.TextMessage, []byte(`{"subscribe": "livedata"}`)); err != nil {
		log.Fatal(err)
	}

	// Start listening for WebSocket messages
	go func() {
		for {
			_, message, err := c.ReadMessage()
			if err != nil {
				log.Println(err)
				return
			}

			// Handle the received JSON message
			handleMessage(message, db)
		}
	}()

	// go func() {
	// 	updateInverterConfig(db)
	// }

	// Keep the program running
	select {}
}
```

**Comment by pieterhollander** [comment]
Also for static information, refer to the [Web API docs](https://tbnobody.github.io/OpenDTU-docs/firmware/web_api/)

**Comment by MetaChuh** [comment]
@adalbertgreben 

hi, super dass du's geschafft hast 👍

wenn du lust hast, kannst du auch hier zur vervollständigung dein zuletzt gepostetes code snippet im discord chat einfügen.
hier können es leute, die ähnliche fragen haben, wiederfinden, in discord geht's unwiederfindbar unter.
(egal in welchem kanal du es postest 😉)

kannst es auch kombinieren, zb. wie zuletzt im discord chatten, dann hier die für dich relevanten lösungen als nachschlagewerk posten.

greetings
